runtime/drivers/duckdb/duckdb.go

```diff
@@ -409,3 +439,126 @@ func (c *connection) checkErr(err error) error {
 	}
 	return err
 }
+
+// Periodically collects stats using pragma_database_size() and emits as activity events
+func (c *connection) periodicallyEmitStats(d time.Duration) {
+	if c.config.Activity == nil {
+		// Activity client isn't set, there is no need to report stats
+		return
+	}
+
+	statTicker := time.NewTicker(d)
+	for {
+		select {
+		case <-statTicker.C:
```

Comment:
Can you throw in a measure for the size observed on the file system using `c.EstimateSize()`? Just in case DuckDB's metadata isn't completely honest about things

Reply:
Added `duckdb_estimated_size_bytes` metric for this purpose